### PR TITLE
fix update-codegen.sh lints

### PR DESCRIPTION
**Pull request by BenTheElder**
cc @krzyzacy ref https://github.com/kubernetes/test-infra/pull/9380

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -17,8 +17,8 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
-${CODEGEN_PKG}/generate-groups.sh "deepcopy,defaulter,client,lister,informer" \
+SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")/.."
```
Comment (BenTheElder) [inline hack/update-codegen.sh]: most of the test-infra scripts use `$(git rev-parse --show-toplevel)` which has some up and downsides. we should maybe standardize on one of these /shrug

someone should write a good, portable bash library to get the path to the running script :^)

**Comment by BenTheElder** [comment]
👍 Let me know if that fixes https://github.com/kubernetes/test-infra/pull/9380 :-)

**Comment by krzyzacy** [comment]
nope, same error, maybe it doesn't work on mac 🙃

**Comment by BenTheElder** [comment]
`://Users/senlu/work//bin/deepcopy-gen` kinda looks like a GOPATH, is that messed up maybe?

**Comment by krzyzacy** [comment]
```
senlu$ echo $GOPATH
:/Users/senlu/work/
```
should be correct?

it works on my workstation

**Comment by stevekuznetsov** [comment]
This is what I get for cargo-culting :|

**Comment by BenTheElder** [comment]
/shrug Working on getting lints upstream again early in the next release cycle :upside_down_face:

**Pull request merged**